nx-X11/programs/Xserver/dix/events.c

```diff
@@ -2948,7 +2948,11 @@ OtherClientGone(void * value, XID id)
 }
 
 int
+#ifdef NXAGENT_SERVER
+Xorg_EventSelectForWindow(register WindowPtr pWin, register ClientPtr client, Mask mask)
+#else
 EventSelectForWindow(register WindowPtr pWin, register ClientPtr client, Mask mask)
+#endif
```

Comment:
I don't like this one. EventSelectForWindow is called from within ChangeWindowAttributes (in dix/window.c). ChangeWindowAttribute is a method set in the pScreen struct. We do define pScreen->ChangeWindowAttributes in Screen.c and assign nxagentChangeWindowAttributes (Window.c) to it. In nxagentChangeWindowAttributes we call XChangeWindowAttributes after doing some nxagent magic.

So, on what code path is the above EventSelectForWindow function really called insider nxagent? I feel that we should handle this a bit more object oriented, that is: define an nxagentEventSelectForWindow and have that called from whereever. Also, I'd prefer to see dix/events.c patched in a way, that ChangeWindowAttributes calls nxagentEventSelectForWindow if NXAGENT_SERVER is defined, EventSelectForWindow (from X.Org), if not defined.